src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -14,95 +14,95 @@ import { updatePlotConfig } from '../../../redux/actions/componentConfig';
 import ReorderableList from '../../ReorderableList';
 import { getCellSetsHierarchy } from '../../../redux/selectors';
 
+const convertToReorderableListData = (cellClassKeys, selected, hierarchy) => (
+  cellClassKeys.map((key) => {
+    const { name } = _.find(hierarchy, ({ key: currentKey }) => currentKey === key);
+
+    return {
+      key, disabledReorder: !selected, name,
+    };
+  })
+);
+
 const HeatmapMetadataTrackSettings = (props) => {
   const dispatch = useDispatch();
 
   const { componentType } = props;
-  const selectedHierarchy = useSelector(getCellSetsHierarchy());
+  const hierarchy = useSelector(getCellSetsHierarchy());
 
   const selectedTracks = useSelector(
     (state) => state.componentConfig[componentType].config.selectedTracks,
   );
 
-  const getTrackData = () => selectedHierarchy.map(
-    (data) => ({ selected: selectedTracks.includes(data.key), key: data.key }),
-  );
+  const [listData, setListData] = useState([]);
 
-  const isInitialRenderRef = useRef(true);
+  useEffect(() => {
+    if (hierarchy === null || _.isEmpty(hierarchy)) return;
 
-  const [trackData, setTrackData] = useState(getTrackData());
+    const selectedTracksListData = convertToReorderableListData(
+      selectedTracks, true, hierarchy,
+    );
 
-  const getUpdatedTrackData = () => _.unionBy(
-    getTrackData(),
-    trackData,
-    'key',
-  );
+    const unselectedTracks = _.difference(hierarchy.map(({ key }) => key), selectedTracks);
+    const unselectedTracksListData = convertToReorderableListData(
+      unselectedTracks, false, hierarchy,
+    );
 
-  useEffect(() => {
-    // Prevent initial dispatch when object appears
-    if (isInitialRenderRef.current) return;
+    setListData([...selectedTracksListData, ...unselectedTracksListData]);
```

Comment:
I see what you're doing here, but I wonder if there is a way to update only the element that changed, rather than having to recreate the `selectedTracks` from scratch every time. For example, if I enable a track, instead of re-creating the entire `selectedTracks`, can we not just go and change `disabledReorder` for that track to `true`?

It is also true that we won't have many `cellClassKeys` to loop over (we only have custom cell sets, samples and louvain), so there is little value into optimising this.

Reply:
good point, I'll look into improving the rendering

Reply:
Hmm so the thing is that the way we are rendering this is by looking at an array: 
`selectedTracks` which just holds which tracks are enabled, so for example if we `louvain` and `samples` enabled then: `selectedTracks = ['louvain', 'sample'];`

So when a track is hidden (`sample` for example) we do: `selectedTracks = ['louvain']`. Everything with `disabledReorder` and all of that is generated using `selectedTracks`.

(To reorder we just reorder the `selectedTracks`).

I think if we wanted to change this to make order and selection not be dependent on each other we would need to change what is being stored in redux for something more complex, which I am not sure if it is worth it since the primary objective here is creating `selectedTracks` for `populateHeatmapData`.

Reply:
I agree, and considering the fact that this is super small amount of data and there is very little impact on performance, it is not really worth it.

Reply:
you can spend your time better by optimising other things that really do matter